[Enhancement] changed some codes for recent std and refactor of others codes

dudantas opened the pull request:
Refactor codes (added from feature of c++ 11/17/20)

- [ ] Pull request breakdown
- [ ] Documentation of changes

dudantas committed:
Remove all boosts, change somes codes for std and others

Refactor other code and remove all boosts
Refactored IOMap::loadMap function and added parse from rme

dudantas committed:
Sonar fixes, somes codes improvements and others

Fixed a lot of sonar checks
Created new function "getTimeNow" in Game class, for get the real time using std::chrono
Created function "setPosition" for Position class
Removed functions "readAttr" from classes; Teleport, Bed, Container, DepotLocker, Door
The attribute is now read directly from item.cpp, and classes just return virtual functions

IOMap::loadMap
Changed BinaryNode for use "shared_ptr"
Various improvements on map loading, added several sanity checks and other improvements
Removed "setLastErrorString" function
Added instead just the spdlog log
Improved the function of checking teleports

Refactor DBResult function "getNumber"

The function "DBResult::getNumber has been broken into several smaller functions, instead of using a generalized function, we will use a function for each value, with sanity checks and using the new features of c++ to perform the string conversions to the appropriate types (int/uint, time_t/boolean).

With this change, we've prevented data loss and added some sanity checks to ensure that if there is a problem, we have a proper log of what happened.

New c++ functions
DBResult::get8()
DBResult::get16()
DBResult::get32()
DBResult::get64()

DBResult::getU8()
DBResult::getU16()
DBResult::getU32()
DBResult::getU64()

DBResult::getTime()
DBResult::getBoolean()

New Lua functions
result:get8()
result:get16()
result:get32()
result:get64()

result:getU8()
result:getU16()
result:getU32()
result:getU64()

result:getTime()
result:getBoolean()

Change negative comparison to call feature of c++20 "std::cmp_less"

Minimum required version for visual studio

Fixed remaining conflict resolution bugs

Sonar fixes, somes codes improvements and others

Fixed a lot of sonar checks
Created new function "getTimeNow" in Game class, for get the real time using std::chrono
Created function "setPosition" for Position class
Removed functions "readAttr" from classes; Teleport, Bed, Container, DepotLocker, Door
The attribute is now read directly from item.cpp, and classes just return virtual functions

IOMap::loadMap
Changed BinaryNode for use "shared_ptr"
Various improvements on map loading, added several sanity checks and other improvements
Removed "setLastErrorString" function
Added instead just the spdlog log
Improved the function of checking teleports

Refactor DBResult function "getNumber"

The function "DBResult::getNumber has been broken into several smaller functions, instead of using a generalized function, we will use a function for each value, with sanity checks and using the new features of c++ to perform the string conversions to the appropriate types (int/uint, time_t/boolean).

With this change, we've prevented data loss and added some sanity checks to ensure that if there is a problem, we have a proper log of what happened.

New c++ functions
DBResult::get8()
DBResult::get16()
DBResult::get32()
DBResult::get64()

DBResult::getU8()
DBResult::getU16()
DBResult::getU32()
DBResult::getU64()

DBResult::getTime()
DBResult::getBoolean()

New Lua functions
result:get8()
result:get16()
result:get32()
result:get64()

result:getU8()
result:getU16()
result:getU32()
result:getU64()

result:getTime()
result:getBoolean()

Change negative comparison to call feature of c++20 "std::cmp_less"

Minimum required version for visual studio

Fixed remaining conflict resolution bugs

Sonar fixes, somes codes improvements and others

Fixed a lot of sonar checks
Created new function "getTimeNow" in Game class, for get the real time using std::chrono
Created function "setPosition" for Position class
Removed functions "readAttr" from classes; Teleport, Bed, Container, DepotLocker, Door
The attribute is now read directly from item.cpp, and classes just return virtual functions

IOMap::loadMap
Changed BinaryNode for use "shared_ptr"
Various improvements on map loading, added several sanity checks and other improvements
Removed "setLastErrorString" function
Added instead just the spdlog log
Improved the function of checking teleports

Refactor DBResult function "getNumber"

The function "DBResult::getNumber has been broken into several smaller functions, instead of using a generalized function, we will use a function for each value, with sanity checks and using the new features of c++ to perform the string conversions to the appropriate types (int/uint, time_t/boolean).

With this change, we've prevented data loss and added some sanity checks to ensure that if there is a problem, we have a proper log of what happened.

New c++ functions
DBResult::get8()
DBResult::get16()
DBResult::get32()
DBResult::get64()

DBResult::getU8()
DBResult::getU16()
DBResult::getU32()
DBResult::getU64()

DBResult::getTime()
DBResult::getBoolean()

New Lua functions
result:get8()
result:get16()
result:get32()
result:get64()

result:getU8()
result:getU16()
result:getU32()
result:getU64()

result:getTime()
result:getBoolean()

Change negative comparison to call feature of c++20 "std::cmp_less"

Minimum required version for visual studio

Fixed remaining conflict resolution bugs

dudantas committed:
Change from malloc to new (from c-style to c++ )

dudantas committed:
Fix sonar checks

dudantas committed:
Small rework in compression and enableCompression functions

dudantas committed:
Fix network max size

dudantas committed:
Fix sonar check

dudantas committed:
Extract unjustified kills in new function for should not be nested too deeply

dudantas committed:
Fix bug sonar, set negative comparison for std::cmp_greater

dudantas committed:
Refactor Outfits::loadFromXml to reduce its cognitive complexity

dudantas committed:
Extract row[result] in own variable

dudantas committed:
Add new header to all source files

dudantas committed:
Fix sonar checks

dudantas committed:
Fix sonar critical and major code smells

dudantas committed:
Fix linux compilation warnings and errors

dudantas committed:
Squashed commit of the following:

commit 6ad0268fa44156cd57b0791abcb09d4a28206472
Author: Beats <61994374+beats-dh@users.noreply.github.com>
Date:   Tue Jun 14 11:07:29 2022 -0300

    Disabled install LuaJit for processor ARM (#416)

    Disabled installation of LuaJit by vcpkg.json, solving the problem of compatibility with ARM processors.
    The installation of LuaJit will have to be done by apt Linux with the command:

    sudo apt install libluajit-5.1-dev

dudantas committed:
Fix sonar check related to localtime using

Call getTime from Game class

Fix linux localtime to localtime_r and windows to localtime_s

dudantas committed:
Fix sonar check and others improvements

Created new class ItemReadMapAttributes (derived from Item)
To break the function, which was very large, in smaller functions, improving the reading and organization of the file item.cpp

Converted a lot of "iterator" pointer implicit conversion to "std::to_address"
Moved a lot of functions from ".h" to ".cpp", to avoid that when modifying an include file we have to recompile a large number of files
Fixed various init-statement sonar checks
Converted a lot of "find" and similiar map functions to "contains"
Converted a lot of "std::find" to"std::ranges::find"
Converted a lot of "std::any_of" to"std::ranges::any_of"
Converted a lot of "std::sort" to"std::ranges::sort"
Converted a lot of "std::for_each" to"std::ranges::for_each"
Converted a lot of "std::shuffle" to"std::ranges::shuffle"
Converted a lot of "std::count" to"std::ranges::count"
Converted a lot of "std::find_if" to"std::ranges::find_if"

Added default operator <=> in position.h instead of manual verifications
Renamed various variables and added structured binding for some std::map loops (iobestiary.cpp, protocolgame.cpp)

Indentation of various functions

dudantas committed:
Fix sonar security hotspots

Change from strlen to custom implementation strnlength
And change from sprintf to snprintf (in tools.cpp) for secure printf

dudantas committed:
Added a using-declaration to inherit the constructors of "Script"

Added a using-declaration to inherit the constructors of "Script", and removed the manually declaration

dudantas committed:
Structured binding of movements.cpp

dudantas committed:
Fix sonar check

Added some variables and functions to const
Enhancement in various loop for
Removed unused variable "vocationId" in function "sendHighscores"
Small enhancement in function "Game::getTime" for use custom time
Removed operator == from Position class
Extracted function readAttributeTileFlags and readAttributeTileItem from IOMap tile parse, using std::tuple

Others sonar check fixes

dudantas committed:
Add default case again

dudantas committed:
Fix sonar code smells

Added const to somes "auto" loops
Added string_view to ParsePlayerCustomFlagMap
Removed "vocation" unused variable in Game::playerHighscores
Some others minor fixes

dudantas committed:
Extract createMapItem in a separated function on map.cpp

The complexity of the function is reduced and code organization/reading is improved.

dudantas committed:
Use the transparent equality "std::string_view for string container

dudantas committed:
Fix sonar critical and major code smells

Set FileHandle for use unique_ptr instead of new gross
Extract std::function from Game::playerHightScores in a separated method named Game::playerSendHighscoreTask for reduce the complexity of function
Added const and reference for some functions
Added using instead of typedef for CustomAttributeMap in item.h
Removed "using NameMap" from unordered multimap of nameToItems in items.h
Added std::string_view for isNumber and isAlpha functions of tools.cpp

guispiller commented:
Made some tests, and this is what I find out so far:
- ~~Look on houses door doesn't give information about the house, just say open/closed door;~~
- ~~Any player can open/close any house doors;~~
- ~~Market doesn't open when clicking;~~
- ~~Teleports aren't working on stepIn;~~
- ~~GOD talkaction /c isn't finding players by their names.~~

> -- edit 1, find more issues:

- ~~Reload scripts is showing warning duplicate event in all scripts;~~
- ~~Inventory items disappear when you logout;~~
- ~~Imbuements are not being applied. You use the imbuement machine, it succesfully apply the imbuement, but the weapon is not imbued;~~
- ~~Familiar spells are summoning monsters with wrong sprite;~~

dudantas committed:
Remove std::ranges::all_of and add for loop

dudantas committed:
Fix some erros and adjust others things

dudantas committed:
Fix some erros and adjust others things related to shared and unique_ptr

dudantas committed:
Fix compilation

dudantas committed:
Fix compilation

dudantas committed:
Fix compilation

dudantas committed:
Fix movement warning on reload scripts

dudantas committed:
Fix reported errors and others improvements

Fixed error on class ItemReadMapAttributes for unserialization of items
Fixed error for not saving player items
Renamed "attributes" to "itemAttributesPtr" for shared_ptr of ItemAttributes
Indentation of IOLoginData and extracted OpenContainer codes in own functions for more organized code and better reading

dudantas committed:
Log improvement

If the player does not have the backpack, it will only inform if it is running in debug

dudantas committed:
Fix some getU sizes

dudantas committed:
Add somes logs to IOLoginData

dudantas committed:
Fix database get types

dudantas committed:
Add lexical_cast.hpp file for lexical casting (number to string and string to number)

dudantas committed:
Change to shared_ptr and BinaryNode reference and others things

dudantas committed:
Rename variables

dudantas committed:
Fix sonar bugs, removed reference from nullpointer

dudantas committed:
Remove unused function

dudantas committed:
Fix not saving backpack on logout

dudantas committed:
Change for use LexicalCast class

dudantas committed:
Make private variables "errorCode, file and cachePtr"

Added public functions for access this instead of directly protected functions

dudantas committed:
Fix wrong sizes

dudantas committed:
Some sonar fixes and others adjustments

beats-dh committed:
Replace all std::unordered with phmap::flat_hash

dudantas committed:
Add const to some functions and move functions corpse from .h to .cpp

dudantas committed:
Merge branch 'main' into remove-all-boosts

dudantas committed:
Change from c-style array to std::vector

dudantas committed:
Add clear vector to getChild and advance functions

dudantas commented:
> Made some tests, and this is what I find out so far:
> 
> * ~Look on houses door doesn't give information about the house, just say open/closed door;~
> * ~Any player can open/close any house doors;~
> * ~Market doesn't open when clicking;~
> * ~Teleports aren't working on stepIn;~
> * ~GOD talkaction /c isn't finding players by their names.~
> 
> > -- edit 1, find more issues:
> 
> * ~Reload scripts is showing warning duplicate event in all scripts;~
> * ~Inventory items disappear when you logout;~
> * ~Imbuements are not being applied. You use the imbuement machine, it succesfully apply the imbuement, but the weapon is not imbued;~
> * ~Familiar spells are summoning monsters with wrong sprite;~

Thanks for testing and reporting. Fixed!

dudantas committed:
Remove unecessary clear cache

dudantas committed:
Add const for some functions, move from .h to .cpp and change from c-style to std::vector

dudantas committed:
Remove duplicated fireStepEvent and executeStep

dudantas committed:
Merge branch 'main' into remove-all-boosts

dudantas committed:
Merge branch 'main' into remove-all-boosts

dudantas committed:
Fix compilation ,remove libzippp

dudantas committed:
Merge branch 'main' into remove-all-boosts

dudantas committed:
Merge branch 'main' into remove-all-boosts

dudantas committed:
Fix compilation

dudantas committed:
Fix compilation

dudantas committed:
Add protobuf::libprotoc and libprotobuf-lite, update cmake version and add required libs

dudantas committed:
Try to fix windows build

dudantas committed:
Fix version

dudantas committed:
Remove builtin baseline

dudantas committed:
add vcxproj to gitignore

dudantas committed:
Merge branch 'main' into remove-all-boosts

dudantas committed:
Update vcpkg commit id for fix gha build

dudantas committed:
Set protobuf min protoc version to 3021003

dudantas committed:
Fix equip/deequip transformation bug

dudantas committed:
Fix showdescription on look for equip and deequip item

dudantas committed:
Change vcxproj location

dudantas committed:
Fix memory leak

dudantas committed:
Fix market offer crash

dudantas committed:
Fixed buying more than 1 item with only one free slot

dudantas committed:
Add player say with npc exhaustion

dudantas committed:
Fix cloning items on depot pages

dudantas committed:
Merge branch 'main' into remove-all-boosts

dudantas committed:
Fix int sizes

dudantas committed:
Merge branch 'main' into remove-all-boosts

dudantas committed:
Fix memory leak

dudantas committed:
Fix lion target

dudantas committed:
Fix imbuement decay time

dudantas committed:
Remove return

dudantas committed:
Fix imbuement not decaying time after second imbuement on login

dudantas committed:
Fix save and open container bug and add logs to Item::readAttr function

dudantas committed:
Introduction of the map in "kmap" format using flatbuffers

 Add flatbuffers lib

dudantas committed:
Add sanity checks

dudantas committed:
Add complete map reading (some adjustments are needed)

dudantas committed:
Merge branch 'main' into remove-all-boosts

dudantas committed:
Add map loading with flatbuffers

dudantas committed:
Fix namespace and filenames

dudantas committed:
Add new kmap loader

dudantas committed:
Some map fixes

dudantas committed:
Fix map loading

dudantas committed:
Fix some errors related to kmap loading

Remove waypoints
Fixed loading kmap items from houses and containers
Fixed items duplication on loading
Reword town class and passed function body to .cpp file

dudantas committed:
Fix

dudantas committed:
Some map load fixes

dudantas committed:
Remove not needed

dudantas committed:
Some map loading fixes

dudantas committed:
Some indent

dudantas committed:
Merge branch 'main' into remove-all-boosts

dudantas committed:
Merge branch 'main' into remove-all-boosts

dudantas committed:
Fixes

dudantas committed:
Some fixes

dudantas committed:
Fixes compilation

dudantas committed:
Merge branch 'main' into remove-all-boosts

dudantas committed:
Fix some MoveEvent conflict

dudantas committed:
Fix some things

dudantas committed:
Some fixes

dudantas committed:
Remove not used files and fix map reading

dudantas committed:
Some fixes

dudantas committed:
Fix some things

dudantas committed:
Added log to print initial function

dudantas committed:
Fixing headers

dudantas committed:
Some sonar fixes

dudantas committed:
Change from switch to unordered_map

Created a dictionary with std::unordered_map that maps AttrTypes_t values to attribute reading functions. That way you can remove the switch and the multiple ifs, and use just one function call to read the attribute.

Since std::function is a function wrapper class, it allows you to store any function or member function, with any signature, in a std::unordered_map. This allows you to map AttrTypes_t values to attribute reading functions simply and quickly.

Also, by using a single function to read the attributes, the code is easier to maintain since all the reading operations are concentrated in one place.

dudantas committed:
Merge branch 'main' into remove-all-boosts

dudantas committed:
Fix container loading and sonar check

dudantas committed:
Add code

dudantas committed:
Fix compilation

dudantas committed:
Merge branch 'main' into remove-all-boosts

dudantas committed:
Fix loading items

dudantas committed:
Some sonar fixes and others reworks
